resolver/type_syntax.cc

```diff
@@ -506,6 +518,18 @@ core::TypePtr interpretTCombinator(core::MutableContext ctx, ast::Send *send, co
                 e.setHeader("Only top-level T.self_type is supported");
             }
             return core::Types::untypedUntracked();
+        case core::Names::attachedClass()._id:
+            if (!ctx.owner.data(ctx)->isSingletonClass(ctx)) {
+                if (auto e = ctx.state.beginError(send->loc, core::errors::Resolver::InvalidTypeDeclaration)) {
+                    e.setHeader("`{}` may only be used in a self context", "T.attached_class");
```

Comment:
```suggestion
                    e.setHeader("`{}` may only be used in a singleton class method context", "T.attached_class");
```

(if this is not what you were going for, feel free to disregard the edit. but I found "self" a bit confusing on first read)

Reply:
I like your wording more, thank you!